### ISSUE TITLE
Webview Service Worker Error in registration due to invalid document state (Linux) 

### ISSUE THREAD

**Issue by tritemio** [microsoft/vscode#125993]
Does this issue occur when all extensions are disabled?: No

- VS Code Version:
```
Version: 1.57.0
Commit: b4c1bd0a9b03c749ea011b06c6d2676c8091a70c
Date: 2021-06-09T17:18:42.895Z
Electron: 12.0.9
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Linux x64 5.11.0-18-generic snap
```
- OS Version: Ubuntu 21.04

Steps to Reproduce:

1. Update to 1.57
2. GitGraph extension stops working

The gitgraph extension shows this error:

```
Error loading webview: Error: Could not register service workers: InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state..
```

**Comment by mhutchie** [comment]
I'm the author of Git Graph, which is one of the extensions that is affected by this (as my extension leverages Visual Studio Code Webview's for it's core user interface). This Service Worker error occurs within Visual Studio Code's Webview implementation, so it affects all extensions that use Webviews. It prevents the Webview from loading, for all the affected users.

https://github.com/mhutchie/vscode-git-graph/issues/508 includes an image of a stack trace of this error in the first comment.

This bug first appeared in Visual Studio Code 1.56.0, and only occurred for numerous Linux users (but definitely not all). Reports of this issue lasted about a week or so, so I assumed it had been fixed in one of the 1.56.x patches. However, with the release of Visual Studio Code 1.57.0, I've now had reports that this issue is occurring again.

Looping in @mjbvz - Are you able to provide some assistance?

**Comment by tritemio** [comment]
For information: restarting the PC solved the issue for me ATM. 

So, it may be related to some sort of cache that is cleared after reboot? 

This issue appeared for me twice, every time after a vs code update. The first time updating from 1.55 to 1.56 and now updating from 1.56 to 1.57. I don't know if the fix on 1.56 happened after a reboot as well.

**Comment by sketchbuch** [comment]
I also have this issue on Linux, be good to get a fix soon

**Comment by mjbvz** [comment]
Duplicate of #122951

**Comment by grandrew** [comment]
Rebooting is not an option for me, and I love the Git Graph extension. As well as ability to view images and other neats within vscode tabs. Please consider re-opening and fixing. Thanks

**Comment by eduncan911** [comment]
@mjbvz please reopen this issue.  @deepak1556 just stated in #122951 [that Linux users are a separate user](https://github.com/microsoft/vscode/issues/122951#issuecomment-884393176).

**Comment by eduncan911** [comment]
Copying my comment [from the "Windows" thread](https://github.com/microsoft/vscode/issues/122951#issuecomment-884364791)...

Running `vscode 1.58.1` on Linux.

Multiple extensions that use WebView all error with:

    Error loading webview: Error: Could not register service workers: InvalidStateError: 
    Failed to register a ServiceWorker: The document is in an invalid state..

`code --no-sandbox` does not help, same errors.

However, **_changing my `--user-data-dir` (that forced a re-install of all extensions) did make it work!_**  

Obviously, that is not preferred due to many years of customizing - that I do not want to start all over with.

```
Ubuntu 21.04
---
Version: 1.58.1
Commit: 2aeda6b18e13c4f4f9edf6667158a6b8d408874b
Date: 2021-07-13T06:20:02.397Z
Electron: 12.0.13
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Linux x64 5.11.0-7620-generic snap
```

```
$ code --verbose .
[1580288:0721/131142.013405:ERROR:appcenter_api.cc(55)] expecting appcenter url prefix
/snap/core/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so)
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
(code:1580288): Gtk-WARNING **: 13:11:42.028: Failed to parse /home/eric/.config/gtk-3.0/settings.ini: Key file does not start with a group
[main 2021-07-21T17:11:42.256Z] Sending env to running instance...
[main 2021-07-21T17:11:42.345Z] Sent env to running instance. Terminating...
[main 2021-07-21T17:11:42.345Z] Lifecycle#kill()
```

(it's installed via Snap)

Developer Tools **_before_** opening a file that uses webview has this error:

```
(node:1580425) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. 
Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.(Use `code --trace-deprecation ...` 
to show where the warning was created) (at writeOut (internal/process/warning.js:43:3))
m @ console.ts:137
```

Developer Tools **_after_** I open a file that fails webview:

```
Webview fatal error: Error: Could not register service workers: InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state..
(anonymous) @ main.js:631
async function (async)
(anonymous) @ main.js:629
(anonymous) @ host.js:27
notificationsAlerts.ts:40 

Error loading webview: Error: Could not register service workers: InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state..
onDidChangeNotification @ notificationsAlerts.ts:40
(anonymous) @ notificationsAlerts.ts:26
fire @ event.ts:583
addNotification @ notifications.ts:206
error @ notificationService.ts:92
(anonymous) @ baseWebviewElement.ts:210
(anonymous) @ webviewElement.ts:168
```

Additional info:

```
$ code --list-extensions
austin.code-gnu-global
be5invis.vscode-icontheme-nomo-dark
budparr.language-hugo-vscode
cssho.vscode-svgviewer
DavidAnson.vscode-markdownlint
dbaeumer.vscode-eslint
DigitalBrainstem.javascript-ejs-support
DotJoshJohnson.xml
dphans.micropython-ide-vscode
eg2.vscode-npm-script
equinusocio.vsc-material-theme-icons
esbenp.prettier-vscode
fivethree.vscode-hugo-snippets
golang.go
hediet.vscode-drawio
jamesmaj.easy-icons
janisdd.vscode-edit-csv
jdinhlife.gruvbox
jebbs.plantuml
JuanBlanco.solidity
magicstack.MagicPython
minhthai.vscode-todo-parser
mitaki28.vscode-clang
ms-azuretools.vscode-docker
ms-python.python
ms-python.vscode-pylance
ms-toolsai.jupyter
ms-vscode.cpptools
ms-vscode.Theme-MarkdownKit
ms-vsliveshare.vsliveshare
ms-vsliveshare.vsliveshare-audio
ms-vsliveshare.vsliveshare-pack
mushan.vscode-paste-image
mutantdino.resourcemonitor
nopeslide.vscode-drawio-plugin-mermaid
phplasma.csv-to-table
PKief.material-icon-theme
platformio.platformio-ide
redhat.vscode-yaml
rid9.datetime
Shan.code-settings-sync
smlombardi.theme-tesla
Stephanvs.dot
streetsidesoftware.code-spell-checker
vsciot-vscode.vscode-arduino
yzhang.markdown-all-in-one
```

Here's my `settings.json` that has been scrubbed from sensitive information:

```
// Place your settings in this file to overwrite the default settings
{
  "workbench.colorTheme": "Gruvbox Dark Soft",
  "workbench.activityBar.visible": true,
  //"workbench.iconTheme": "vscode-icons",
  //"workbench.iconTheme": null,
  "editor.cursorBlinking": "solid",
  //"editor.fontFamily": "'Go Mono'",
  "editor.fontLigatures": true,
  "editor.renderIndentGuides": false,
  "editor.minimap.enabled": false,
  "editor.tabSize": 2,
  "editor.stickyTabStops": true,
  "editor.formatOnSave": true,
  "editor.defaultFormatter": null,
  "[javascript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "[jsonc]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "[json]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "editor.codeActionsOnSave": {
    "source.fixAll": true
  },
  "files.insertFinalNewline": true,
  "files.trimFinalNewlines": true,
  "files.trimTrailingWhitespace": true,
  "explorer.openEditors.visible": 0,
  "explorer.confirmDelete": false,
  "explorer.confirmDragAndDrop": false,
  "extensions.autoUpdate": false,
  //"terminal.integrated.fontFamily": "Go Mono, Inconsolata-dz for Powerline",
  "terminal.integrated.lineHeight": 0,
  "window.openFilesInNewWindow": "off",
  "extensions.ignoreRecommendations": true,
  "explorer.compactFolders": false,
  "vim.enableNeovim": true,
  "vim.disableAnnoyingNeovimMessage": true,
  "go.testOnSave": true,
  "go.coverOnSave": true,
  "go.useCodeSnippetsOnFunctionSuggest": true,
  "go.autocompleteUnimportedPackages": true,
  "go.useLanguageServer": true,
  "C_Cpp.intelliSenseEngine": "Default",
  "sync.gist": "(snip)",
  "sync.lastUpload": "2018-07-02T18:38:38.162Z",
  "sync.autoDownload": true,
  "sync.autoUpload": true,
  "sync.lastDownload": "2018-02-19T15:13:34.109Z",
  "sync.forceDownload": false,
  "sync.anonymousGist": false,
  "sync.host": "",
  "sync.pathPrefix": "",
  "sync.quietSync": true,
  "sync.askGistName": false,
  "vsicons.dontShowNewVersionMessage": true,
  "python.pythonPath": "",
  "workbench.startupEditor": "newUntitledFile",
  "python.linting.enabled": true,
  "python.linting.pylintEnabled": true,
  "python.linting.pylamaEnabled": false,
  "python.linting.pycodestyleEnabled": false,
  "python.linting.flake8Enabled": false,
  "sync.removeExtensions": true,
  "sync.syncExtensions": true,
  "extensions.autoCheckUpdates": false,
  "update.showReleaseNotes": false,
  "telemetry.enableCrashReporter": false,
  "telemetry.enableTelemetry": true,
  "npm.fetchOnlinePackageInfo": true,
  "workbench.enableExperiments": false,
  "workbench.settings.enableNaturalLanguageSearch": false,
  "sync.forceUpload": true,
  "zenMode.hideTabs": false,
  "dateTime.use24HourClock": false,
  "dateTime.showAMPM": true,
  "resmon.show.cpufreq": false,
  "resmon.show.cputemp": true,
  "resmon.show.cpuusage": false,
  "resmon.show.mem": false,
  "debug.console.fontSize": 12.5,
  "markdown.preview.fontSize": 12.5,
  "editor.fontSize": 12.5,
  "terminal.integrated.fontSize": 12.5,
  "workbench.colorCustomizations": {
    "[Tesla]": {
      "activityBar.background": "#232323EE",
      "editor.inactiveSelectionBackground": "#3d3d3d",
      "editor.selectionBackground": "#636363",
      "editor.selectionHighlightBackground": "#505050",
      "scrollbar.shadow": "#23232300",
      "sideBar.background": "#232323EE",
      "statusBar.background": "#23232300",
      //"statusBar.background": "#000000FF",
      //"statusBar.border": "#3B3B3B",
      "statusBar.foreground": "#969696",
      "statusBar.debuggingBackground": "#511f1f",
      "statusBar.debuggingBorder": "#8b1818",
      "statusBar.debuggingForeground": "#FFFFFF",
      "tab.activeBorderTop": "#3d3d3d"
    },
    "terminal.background": "#282828",
    "terminal.foreground": "#D5C4A1",
    "terminalCursor.background": "#D5C4A1",
    "terminalCursor.foreground": "#D5C4A1",
    "terminal.ansiBlack": "#282828",
    "terminal.ansiBlue": "#83A598",
    "terminal.ansiBrightBlack": "#665C54",
    "terminal.ansiBrightBlue": "#83A598",
    "terminal.ansiBrightCyan": "#8EC07C",
    "terminal.ansiBrightGreen": "#B8BB26",
    "terminal.ansiBrightMagenta": "#D3869B",
    "terminal.ansiBrightRed": "#FB4934",
    "terminal.ansiBrightWhite": "#FBF1C7",
    "terminal.ansiBrightYellow": "#FABD2F",
    "terminal.ansiCyan": "#8EC07C",
    "terminal.ansiGreen": "#B8BB26",
    "terminal.ansiMagenta": "#D3869B",
    "terminal.ansiRed": "#FB4934",
    "terminal.ansiWhite": "#D5C4A1",
    "terminal.ansiYellow": "#FABD2F"
  },
  "editor.suggest.showIcons": false,
  "workbench.editor.showIcons": true,
  //"workbench.iconTheme": "material-icon-theme",
  //"workbench.iconTheme": null,
  "breadcrumbs.enabled": false,
  "window.menuBarVisibility": "toggle",
  "cSpell.userWords": [
    "snipped"
  ],
  "files.associations": {
    "**/*.yml": "ansible"
  },
  "go.formatTool": "gofmt",
  "go.coverageOptions": "showUncoveredCodeOnly",
  "go.coverOnSingleTest": true,
  "go.coverOnSingleTestFile": true,
  "go.coverShowCounts": true,
  "terminal.integrated.tabs.enabled": true,
  "javascript.updateImportsOnFileMove.enabled": "always",
  "window.zoomLevel": 1,
  "hediet.vscode-drawio.customFonts": [
    "Go",
    "Go Medium",
    "Go Mono",
    "Go, Bold",
    "Go Mono, Bold",
    "Go Medium, Bold"
  ],
  "hediet.vscode-drawio.local-storage": "(snip)",
  "TodoParser": {
    "folderExclude": ["node_modules", ".vscode"],
    "showInProblems": true,
    "markers": [
      ["TODO", "Warning"],
      ["FIXME", "Warning"]
    ],
    "autoDefaultMarkers": true
  },
  "hediet.vscode-drawio.theme": "min",
  "hediet.vscode-drawio.codeLinkActivated": false,
  "docker.showStartPage": false,
  "[markdown]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "emmet.includeLanguages": {
    "ejs": "html"
  },
  "workbench.editorAssociations": {
    "*.ipynb": "jupyter.notebook.ipynb"
  },
  "workbench.iconTheme": "eq-material-theme-icons-darker",
  "typescript.updateImportsOnFileMove.enabled": "always",
  "files.exclude": {
    "**/node_modules": true
  },
  "redhat.telemetry.enabled": false
}
```

I tried to downgrade to VSCode 1.55; but, when opening a file now for webview, it doesn't seem to trigger the extensions any longer (something broke).

**Comment by deepak1556** [comment]
Please provide the trace following the steps below:

* Open vscode like `code --enable-tracing="navigation,net,loading,rail,ServiceWorker,disabled-by-default-loading" --trace-startup-file="/tmp/chrometrace.log"`
* Perform the steps that leads to the the service worker error
* Quit the application
* Attach the `chrometrace.log` file generated at `/tmp`

**Comment by tamuratak** [comment]
We can see that all the reported issues happen with the snap version of VS Code. It is an unofficial build. It would be a fix to install the official build of VS Code.

**Comment by eduncan911** [comment]
> > OS: Linux x64 5.11.0-18-generic snap
> 
> We can see that all the reported issues happen with the snap version of VS Code. It is an unofficial build. It would be a fix to install the official build of VS Code.

Right at the top of the Installation page, it says:

> Snap#
 > Visual Studio Code is officially distributed as a Snap package in the Snap Store
> 

https://code.visualstudio.com/docs/setup/linux

![Screenshot_20210721-212721](https://user-images.githubusercontent.com/104980/126578923-cb94dc68-d220-42f3-899f-0dd9cf88bead.png)

**Comment by tamuratak** [comment]
> I think that's incorrect.

Thank you for your correction.

**Comment by eduncan911** [comment]
> Please provide the trace following the steps below:
> 
> * Open vscode like `code --enable-tracing="navigation,net,loading,rail,ServiceWorker,disabled-by-default-loading" --trace-startup-file="/tmp/chrometrace.log"`
> * Perform the steps that leads to the the service worker error
> * Quit the application
> * Attach the `chrometrace.log` file generated at `/tmp`

Doesn't look like `--enable-tracing` nor `--trace-startup-file` are valid options.

    $ code --enable-tracing="navigation,net,loading,rail,ServiceWorker,disabled-by-default-loading" --trace-startup-file="/tmp/chrometrace.log" .
    Warning: 'enable-tracing' is not in the list of known options, but still passed to Electron/Chromium.
    Warning: 'trace-startup-file' is not in the list of known options, but still passed to Electron/Chromium.

However, there does seem to be something in the log file (see attached).

[chrometrace.log](https://github.com/microsoft/vscode/files/6865856/chrometrace.log)

**Comment by deepak1556** [comment]
Thanks for the log, you can ignore the warning. They just mean the flags are not respected by the application but will be sent to the underlying runtime.

The log you have captured does not include any loading data, can you attach a gif recording of the steps that you performed. It should be something like the following where a simple launch of the application should generate a good amount of data.

[recording.gif.zip](https://github.com/microsoft/vscode/files/6866458/recording.gif.zip)

**Comment by eduncan911** [comment]
Before doing the recording, I went through the steps again - and this time the log file seems to be much larger.  Open vscode, open the file that produces the error, close vscode.  Not sure why the first log was so small.

[chrometrace.log](https://github.com/microsoft/vscode/files/6866516/chrometrace.log)

**Comment by deepak1556** [comment]
Thanks for the logs, @mjbvz here is what we know from it. The following is the loading trace comparison of draw-io extension from my machine and from the issue reporter. I will skip to the part which starts loading the webview container.

1) The navigation to `vscode-webview://{{uuid}}/index.html?id=3d61c755-1203-4a27-8874-30a8c77cc9c2&swVersion=2&extensionId=hediet.vscode-drawio&platform=electron&vscode-resource-base-authority=vscode-resource.vscode-webview.net` gets committed on both these traces, which means there is successful navigation to the container startpoint and the render process is locked to this origin.

Working device:

```
Title | RenderFrameHostImpl::DidCommitProvisionalLoad 
-- | --
Category | navigation
User Friendly Category | other
Start | 15,992.164 ms
Wall Duration | 0.793 ms
CPU Duration | 0.793 ms
Self Time | 0.483 ms
CPU Self Time | 0.482 ms
▶Args |  
params | {contents_mime_type: "", did_create_new_entry: false, document_sequence_number: -1642300397, embedding_token: "5672A309BEE3D95750DCA8866CD6EF0B", has_potentially_trustworthy_unique_origin: false, history_list_was_cleared: false, http_status_code: 200, intended_as_new_entry: false, is_overriding_user_agent: false, item_sequence_number: -1642300398, method: "GET", navigation_token: "4ED2F51C423A27DA45E06C18CC2A46A8", origin: "vscode-webview://a3475ae9-bbd7-4d29-97a1-c162a7e22d3f", original_request_url: "", post_id: -1, process lock: "{ vscode-webview://a3475ae9-bbd7-4d29-97a1-c162a7e22d3f/ }", referrer: {policy: 6,            url: ""}, request_id: 80005, should_replace_current_entry: true, should_update_history: true, transition: 3, url: "vscode-webview://a3475ae9-bbd7-4d29-97a1-c162a7e22d3f/index.html?id=a3475ae9-bbd7-4d29-97a1-c162a7e22d3f&swVersion=2&extensionId=hediet.vscode-drawio&platform=electron&vscode-resource-base-authority=vscode-resource.vscode-webview.net", url_is_unreachable: false}
rfh | {browsing_instance_id: 2, frame_tree_node_id: 3, lifecycle_state: "Speculative", origin: "null [internally: (43163F393B9C82B24A1737DE519B042C) anonymous]", parent: {browsing_instance_id: 2,          frame_tree_node_id: 1,          lifecycle_state: "Active",          origin: "vscode-file://vscode-app",          process_id: 6,          render_frame_id: 1,          site_instance_id: 3,          this: "0x1d6376a5a000",          url: "vscode-file://vscode-app/snap/code/70/usr/share/code/resources/app/out/vs/code/electron-browser/workbench/workbench.html"}, process_id: 9, render_frame_id: 5, site_instance_id: 6, this: "0x1d6377e1f000",
```

Error device:

```
Title | RenderFrameHostImpl::DidCommitProvisionalLoad
-- | --
Category | navigation
User Friendly Category | other
Start | 5,114.595 ms
Wall Duration | 3.906 ms
CPU Duration | 0.770 ms
Self Time | 3.607 ms
CPU Self Time | 0.462 ms
▶Args |  
params | {contents_mime_type: "", did_create_new_entry: false, document_sequence_number: -25912278, embedding_token: "A9A1236A1CB9944CF3D18EB533A211CD", has_potentially_trustworthy_unique_origin: false, history_list_was_cleared: false, http_status_code: 200, intended_as_new_entry: false, is_overriding_user_agent: false, item_sequence_number: -25912279, method: "GET", navigation_token: "57F47682D127BEC4D9865DBD5B003000", origin: "vscode-webview://3d61c755-1203-4a27-8874-30a8c77cc9c2", original_request_url: "", post_id: -1, process lock: "{ vscode-webview://3d61c755-1203-4a27-8874-30a8c77cc9c2/ }", referrer: {policy: 6,            url: ""}, request_id: 853431, should_replace_current_entry: true, should_update_history: true, transition: 3, url: "vscode-webview://3d61c755-1203-4a27-8874-30a8c77cc9c2/index.html?id=3d61c755-1203-4a27-8874-30a8c77cc9c2&swVersion=2&extensionId=hediet.vscode-drawio&platform=electron&vscode-resource-base-authority=vscode-resource.vscode-webview.net", url_is_unreachable: false}
rfh | {browsing_instance_id: 2, frame_tree_node_id: 3, lifecycle_state: "Speculative", origin: "null [internally: (196D83E2E26DCF81313630615C6AFFF6) anonymous]", parent: {browsing_instance_id: 2,          frame_tree_node_id: 1,          lifecycle_state: "Active",          origin: "vscode-file://vscode-app",          process_id: 7,          render_frame_id: 1,          site_instance_id: 3,          this: "0x3ec986f27200",          url: "vscode-file://vscode-app/snap/code/70/usr/share/code/resources/app/out/vs/code/electron-browser/workbench/workbench.html"}, process_id: 10, render_frame_id: 5, site_instance_id: 6, this: "0x3ec9870b1400", url: ""}
```

2) Next we get a resource request for `"vscode-webview://{{uuid}}/host.js"` which also gets completed for both scenarios

3) After which get resource request for `"vscode-webview://{{uuid}}/main.js"`, this is the file which tries to register the service worker https://github.com/microsoft/vscode/blob/c1b4cf00b93a83f810055b52ba2ceb08be1432a1/src/vs/workbench/contrib/webview/browser/pre/main.js#L207-L240 

The resource request to `main.js` completes for both scenarios, but the difference comes in the `navigator.serviceWorker.register` call. For the successful scenario we pass the security checks in blink and continue with service worker loading as you can see below

<img width="1671" alt="Screen Shot 2021-07-23 at 11 55 11 AM" src="https://user-images.githubusercontent.com/964386/126828614-c6a13245-fe44-4aa4-b7a3-d175d6d9d960.png">


But for the failure case we are hitting this condition https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/service_worker/service_worker_container.cc;l=312-319 which is the DOM exception the user receives in the end.

Based on some early debugging the provider can be empty under the following conditions:

1) If the page loading the service worker is being navigated away during the registration step
2) If the service worker is registered inside an iframe then the provider can be empty if the sandbox property of the frame is changed dynamically.

From the trace we are definitely not hitting 1) and based on https://github.com/microsoft/vscode/blob/c1b4cf00b93a83f810055b52ba2ceb08be1432a1/src/vs/workbench/contrib/webview/browser/webviewElement.ts#L357-L376 we only edit the sandbox properties during the webview container creation and not after that, and this steps happens way before the `main.js` file gets requested.

So we are hitting some other edge case that is not covered above, will ping the chromium team to understand further or if we can get better debug logs.

**Comment by eduncan911** [comment]
As a reminder from my earlier debugging, when I change the `--user-data-dir` to a new empty directory (and vscode re-downloads all extensions), it works fine.

**Comment by deepak1556** [comment]
@eduncan911 can you try the following and see if you still get the webview error in each case.

1) Disable all extensions except for `draw-io` 
2) Remove `~/.config/Code/Code Cache`
3) Remove `~/.config/Code/Service Worker`

**Comment by eduncan911** [comment]
Disabling all extensions, and only enabling Draw-io worked.  Went to re-enable everything and it continues to work.  However, several of my preferences were reverted.

I have not switched to use MS sync yet - I am still using an old gist-sync method for my settings.  Therefore, I can see everything that changes - and roll back.

I cannot see anything that would have caused this bug to happen.  

Here are the changes to my preferences for when I disabled all extensions:

``` diff
// Place your settings in this file to overwrite the default settings
{
  "workbench.colorTheme": "Gruvbox Dark Soft",
  "workbench.activityBar.visible": true,
  //"workbench.iconTheme": "vscode-icons",
  //"workbench.iconTheme": null,
@@ -166,7 +165,7 @@
  "go.coverShowCounts": true,
  "terminal.integrated.tabs.enabled": true,
  "javascript.updateImportsOnFileMove.enabled": "always",
  "window.zoomLevel": 2,
  "window.zoomLevel": 1,
  "hediet.vscode-drawio.customFonts": [
    "Go",
    "Go Medium",
```

Here are the changes to extensions when I re-enabled most of my extensions:

```
@@ -49,6 +49,16 @@
    "publisher": "phplasma",
    "version": "1.2.3"
  },
  {
    "metadata": {
      "id": "7b69248c-488a-4a74-ae29-56822156726f",
      "publisherId": "rid9.datetime",
      "publisherDisplayName": "rid9"
    },
    "name": "datetime",
    "publisher": "rid9",
    "version": "2.0.3"
  },
  {
    "metadata": {
      "id": "3acc06d9-1bc5-49e1-ae61-306118c95228",
@@ -149,6 +159,16 @@
    "publisher": "PKief",
    "version": "4.8.0"
  },
  {
    "metadata": {
      "id": "e741f81d-b7fb-4d6c-8e8d-f78e796cb8ae",
      "publisherId": "dphans.micropython-ide-vscode",
      "publisherDisplayName": "dphans"
    },
    "name": "micropython-ide-vscode",
    "publisher": "dphans",
    "version": "0.0.1"
  },
  {
    "metadata": {
      "id": "d95cb424-7a5a-4e08-9698-107d6fd590cf",
@@ -159,6 +179,16 @@
    "publisher": "jebbs",
    "version": "2.15.1"
  },
  {
    "metadata": {
      "id": "6cf182b2-70cf-433e-97f2-45f414c9ad89",
      "publisherId": "platformio.platformio-ide",
      "publisherDisplayName": "platformio"
    },
    "name": "platformio-ide",
    "publisher": "platformio",
    "version": "2.3.2"
  },
  {
    "metadata": {
      "id": "96fa4707-6983-4489-b7c5-d5ffdfdcce90",
@@ -169,6 +199,26 @@
    "publisher": "esbenp",
    "version": "8.0.1"
  },
  {
    "metadata": {
      "id": "f1f59ae4-9318-4f3c-a9b5-81b2eaa5f8a5",
      "publisherId": "ms-python.python",
      "publisherDisplayName": "ms-python"
    },
    "name": "python",
    "publisher": "ms-python",
    "version": "2021.7.1050252941"
  },
  {
    "metadata": {
      "id": "6efd3abb-3c6e-49c8-b8ce-6df58982255b",
      "publisherId": "mutantdino.resourcemonitor",
      "publisherDisplayName": "mutantdino"
    },
    "name": "resourcemonitor",
    "publisher": "mutantdino",
    "version": "1.0.7"
  },
  {
    "metadata": {
      "id": "25baab03-fe9a-40c7-b683-55c9a0a92480",
@@ -209,6 +259,16 @@
    "publisher": "equinusocio",
    "version": "2.0.10"
  },
  {
    "metadata": {
      "id": "cf5a111e-3bd1-4e32-9526-20ec73780741",
      "publisherId": "vsciot-vscode.vscode-arduino",
      "publisherDisplayName": "vsciot-vscode"
    },
    "name": "vscode-arduino",
    "publisher": "vsciot-vscode",
    "version": "0.4.3"
  },
  {
    "metadata": {
      "id": "f8dbe169-d649-4e4b-adb8-ffa113c1d616",
@@ -319,6 +379,16 @@
    "publisher": "mushan",
    "version": "1.0.4"
  },
  {
    "metadata": {
      "id": "364d2426-116a-433a-a5d8-a5098dc3afbd",
      "publisherId": "ms-python.vscode-pylance",
      "publisherDisplayName": "ms-python"
    },
    "name": "vscode-pylance",
    "publisher": "ms-python",
    "version": "2021.7.4"
  },
  {
    "metadata": {
      "id": "6f5d4f30-67aa-4151-80c0-1bbe8ca5f9db",
```

Again, nothing stands out.

It's working now.

**Comment by deepak1556** [comment]
Thanks, appreciate it! any additional debug info helps.

**Comment by mjbvz** [comment]
@deepak1556 Thanks for investigating!

I agree that we should not be modifying the sandbox attribute or navigating the webview host page while the service worker is loading. We also wait for the the service worker to be registered before loading the extension provided webview content: https://github.com/microsoft/vscode/blob/99c8e788d46c935aecb16129919c6041eed92250/src/vs/workbench/contrib/webview/browser/pre/main.js#L724

**Comment by eduncan911** [comment]
Closed automatically after 7 days of inactivity?

That seems ... A but excessive.

**Comment by deepak1556** [comment]
If anyone finds a reliable way to repro the issue it would help with further debugging, thanks!

**Comment by Omcsesz** [comment]
I experience the same issue on Ubuntu 21.04., with the latest official VSCode version.

**Comment by eduncan911** [comment]
> I experience the same issue on Ubuntu 21.04., with the latest official VSCode version.

@deepak1556 Well, it's happened to me again as well.

It was fixed for about a week; but, vscode just updated to 1.59. Now broken.

```
Version: 1.59.0
Commit: 379476f0e13988d90fab105c5c19e7abc8b1dea8
Date: 2021-08-04T23:13:20.182Z
Electron: 13.1.7
Chrome: 91.0.4472.124
Node.js: 14.16.0
V8: 9.1.269.36-electron.0
OS: Linux x64 5.11.0-7620-generic snap
```

@Omcsesz  The previous fix was to disable all extensions, and re-enable them. (I also had a few restarts of vscode in the middle of all that)

**Comment by c0ppelius** [comment]
I had been squatting 1.53 for about a month because of this error but was forced to upgrade to install a new extension. I found the following. 

What didn't work: 
- disable and re-enable extensions 
- `apt purge code` and re-install

What did work:
- `apt purge code && rm -r ~/.config/Code` and re-install. 

OS: Ubuntu 20.04.2 
VS Code: 1.59

**Comment by sam-dehority** [comment]
@c0ppelius thismethod also worked for me, under the variation 

`snap remove --purge code && rm -r ~/.config/Code`

OS: Ubuntu 21.04
Code: 1.59.1

**Comment by grandrew** [comment]
Was hit by this issue again, on a completely different machine and setup but was able to resolve without rebooting. 

There where stale "code" processes running in the background even though I explicitly clicked "exit" from menu. I just killed them from `htop` one by one, and the problem was gone.

I suspect this happened due to a previous crash (-es?) - I periodically open my laptop and see no vscode windows open, so I just click the icon again and the window re-appears. Looks like the old processes get orphaned.

**Comment by diggityshmoo** [comment]
> Was hit by this issue again, on a completely different machine and setup but was able to resolve without rebooting.
> 
> There where stale "code" processes running in the background even though I explicitly clicked "exit" from menu. I just killed them from `htop` one by one, and the problem was gone.
> 
> I suspect this happened due to a previous crash (-es?) - I periodically open my laptop and see no vscode windows open, so I just click the icon again and the window re-appears. Looks like the old processes get orphaned.

Just wanted to confirm that this also worked for me on VSCode 1.59.1 running on macOS Big Sur 11.5.1 .  Somehow an instance of Code was stuck running in the background even though I had Quit it previously.  Killing all instances of Code and the Code Helper app through the activity monitor and then restarting resolved the problem for me.

**Comment by tritemio** [comment]
I found that the problem ca be fixed without rebooting by removing all the cache folders used by vscode:

- close vscode (make sure no process is running)
- `rm -rf ~/.config/Code/*Cache*`
- open vs code

**Comment by Omcsesz** [comment]
@eduncan911 The problem persists in version 1.60.

**Comment by eduncan911** [comment]
> @eduncan911 The problem persists in version 1.60.

Yeah, a 2nd comment I made later said it's still an issue.

**Comment by PaluMacil** [comment]
Ubuntu 20.04, snap install, Code 1.60.0: confirming it is still a problem. I opened Code which loaded release notes--I assume from an update last I used it, and I have exactly the same issue. From reading through this, I killed the 31 processes related to Code and then as able to successfully load the release notes on next launch from the help menu.

**Comment by eduncan911** [comment]
Ah yes, we forgot to report that as an issue as well:

This error also prevents the Release Notes from displaying after an upgrade.

**Comment by cernec1999** [comment]
I'm on Ubuntu 20.04 with code 1.60.1. I attempted to update the include path for a C++ project. My VSCode was installed through snap. Here is the error:
```
Error loading webview: Error: Could not register service workers: InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state..
```

**Comment by lubomir-brindza** [comment]
> Was hit by this issue again, on a completely different machine and setup but was able to resolve without rebooting.
> 
> There where stale "code" processes running in the background even though I explicitly clicked "exit" from menu. I just killed them from `htop` one by one, and the problem was gone.

You can kill all running vscode processes with `pkill code`.

Ran into this just now; the cause was probably that I was still running 'old' code instances alongside new ones (updating a snap will cause the Ubuntu launcher icon to indicate no instances are running, since the path to the executable has changed). I have no idea whether the 'old' code instance was responsible for a stray headless process, or I just didn't notice an open window in one of my workspaces; but killing them made the issue go away.

**Comment by xscd** [comment]
Some extensions (which depends on webview I assume) don't work. Details of extenions also don't load. Not sure if its related to this issue since they start to work when --no-sandbox flag is passed some comments here say they don't even work with that flag. 

Fedora Linux 35. Visual Studio Code 1.60.1

**Comment by bstivers** [comment]
```pkill code``` from @lubomir-brindza combined with ```rm -rf ~/.config/Code/*Cache*``` from @tritemio  worked for me. 

```
Version: 1.60.2
Commit: 7f6ab5485bbc008386c4386d08766667e155244e
Date: 2021-09-22T12:01:43.795Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.11.0-34-generic snap (Ubuntu 20.04)
```

**Comment by zouyonghao** [comment]
`pkill code` works for me

```
Version: 1.60.2
Commit: 7f6ab5485bbc008386c4386d08766667e155244e
Date: 2021-09-22T12:01:43.795Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 4.15.0-156-generic snap
```

**Comment by Fdawgs** [comment]
Running into this on Windows 10 after updating VS Code from v1.60 to v1.61.

**Comment by 8bitmp3** [comment]
@karasowles @dynamicwebpaige Codespaces has been affected too, can't use notebooks. PTAL 🙏 

> after updating VS Code from v1.60 to v1.61 (x64).

> This appeared after vscode update to 1.16.0 x64 for me today

Likely because of ⬆️ ?

Similar issues: https://github.com/microsoft/vscode/issues/122951 and https://github.community/t/the-user-denied-permission-to-use-service-worker-when-using-codespaces/195324

**Comment by klequis** [comment]
> Was hit by this issue again, on a completely different machine and setup but was able to resolve without rebooting.
> 
> There where stale "code" processes running in the background even though I explicitly clicked "exit" from menu. I just killed them from `htop` one by one, and the problem was gone.

Same here. I was having this error with the Git Lens extension of Ubuntu 20.04.

There were 10 or so processes that said "code". Once killed I was able to activate Git Lens without issue.

**Comment by mattangus** [comment]
> You can kill all running vscode processes with `pkill code`.
> 
> Ran into this just now; the cause was probably that I was still running 'old' code instances alongside new ones (updating a snap will cause the Ubuntu launcher icon to indicate no instances are running, since the path to the executable has changed). I have no idea whether the 'old' code instance was responsible for a stray headless process, or I just didn't notice an open window in one of my workspaces; but killing them made the issue go away.

I have anecdote to support this. I accidentally left my computer on last night with VSCode running. I log in to find the icon showing there are no windows open. This has happened to a different computer along with the webview error. So I left the windows open just in case. Opening a new window causes the webview error to pop up. Here is a screenshot showing the webview markdown preview working on the old process:

![image](https://user-images.githubusercontent.com/5302804/140495657-9bd44bd1-1d5e-4958-8168-66cffda9150a.png)

Clicking on the icon to opens a new window that has the webview error:

![image](https://user-images.githubusercontent.com/5302804/140495925-9beeb0af-51ae-4a84-a06a-887428722f54.png)

Using `alt + tab` to see all open vscode windows (even the ones that are no longer associated with a tray icon), and closing them normally worked for me. This might be better than using `pkill` to shutdown vscode ungracefully.

I can confirm that an update to VSCode was partially the cause. The old window had the following version info:
```
Version: 1.61.2
Commit: 6cba118ac49a1b88332f312a8f67186f7f3c1643
Date: 2021-10-19T14:58:13.605Z
Electron: 13.5.1
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.4.0-89-generic snap
```

The new window has this version info:
```
Version: 1.62.0
Commit: b3318bc0524af3d74034b8bb8a64df0ccf35549a
Date: 2021-11-03T15:18:44.904Z
Electron: 13.5.1
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.4.0-89-generic snap
```

@deepak1556 I have been able to reproduce this bug with the following:
 1. run `sudo snap revert code` (this causes the tray icon to show no windows open)
 2. click the icon to open a new window (or via command line `code <path>`)
 3. The new window should have the webview error
 4. If all windows are closed (checking with `alt + tab` then the issue goes away)
 5. reintroduce it again with `sudo snap revert code --revision <number>` to force the upgrade again with an existing code instance running.

Hope this helps!

**Comment by deepak1556** [comment]
Thanks @mattangus, those steps are really helpful. I can confirm the issue on my end now.

**Comment by danielpenagos** [comment]
`pkill code`  works for me. Thanks

Version: 1.62.1
Commit: f4af3cbf5a99787542e2a30fe1fd37cd644cc31f
Date: 2021-11-05T10:56:50.217Z
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.11.0-40-generic snap

**Comment by eleach** [comment]
I'm on Ubuntu 20.04, vsCode 1.62.2, installed with snap.

The tritemio steps work very briefly for me and then the same error happens over and over again. Is there any chance this will ever be fixed? Sorry I'm not very helpful here. This is completely ruining my vscode experience. (I wanted to try copilot.) I'm going to try downgrading to 1.55.2 and see what happens.

**Comment by flavianojs** [comment]
I just found out that killing all vscode jobs through the activity monitor also solves the problem.
In this way, one does not need to restart the machine like I was doing before.

Ubuntu 18.04.6 LTS
```
Version: 1.62.3
Commit: ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
Date: 2021-11-17T08:00:36.721Z
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.4.0-90-generic snap
```

**Comment by strowk** [comment]
I am getting this error after running `git rebase -i`, because I have set `git config --global core.editor "code --wait"` , but now I have to switch back to vim. This is so annoying. Issues like this make me want to disable autoupdates, vscode is so unstable lately. 
Ubuntu 20.04, vscode 1.62.3 

Restarting vscode seems to help sometimes, but after some time, error is back again.

**Comment by mjunior** [comment]
> `pkill code` works for me. Thanks
> 
> Version: 1.62.1 Commit: [f4af3cb](https://github.com/microsoft/vscode/commit/f4af3cbf5a99787542e2a30fe1fd37cd644cc31f) Date: 2021-11-05T10:56:50.217Z Electron: 13.5.2 Chrome: 91.0.4472.164 Node.js: 14.16.0 V8: 9.1.269.39-electron.0 OS: Linux x64 5.11.0-40-generic snap

Only pkill works for me.

Version: 1.62.3
Commit: ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
Date: 2021-11-17T08:00:36.721Z
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.11.0-38-generic snap

**Comment by salouri** [comment]
> I also have this issue on Linux, be good to get a fix soon.
> 
> Update: Rebooting fixed the issue, thanks @tritemio

rebooteed what? vscode or OS ?

**Comment by sketchbuch** [comment]
OS

**Comment by sketchbuch** [comment]
Still having this issue: v1.63.2

**Comment by rjmsilveira** [comment]
For me, what worked was to delete the folder `Service Worker` under `~/.config/Code` and re-opening my Visual Studio and it started working as expected.
Hope it helps someone :)

**Comment by GuillaumeAmat** [comment]
On my end, the issue was that I had it installed from both the repo package (Fedora 35) and the Flathub one.
I guess they have a common library path (or service port) but expect different versions of them.

Removing one of them and restarting the remaining VSCode fixes it for good :ok_hand:

**Comment by bmh129** [comment]
I had same problem as the OP (Webview Service Worker Error...), but on a 2012 Mac Mini running macOS 10.15.7, running the following VSCode from the official website:

Version: 1.63.2 (Universal)
Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
Date: 2021-12-15T09:37:28.172Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 19.6.0

This problem persisted for my setup for all VSCode versions since 1.58.2.  For me, the problem only existed when VSCode was run without elevated privileges.  To restate, the problem did not present if I ran VSCode under sudo.  

I was able to resolve my issue by doing the following:

**Drag the VSCode app to the trash, empty the trash, then, from the terminal, remove the entire folder at ~/Library/Application Support/Code, followed by a reboot, then installation of the latest VSCode.**

**Comment by avarayr** [comment]
Easy repro for Intel Macbook (MacOS 12.1), please try with your OS with appropriate hotkeys.

0. Quit all instances of vscode
1. Open terminal
2. `code-insiders test.ipynb`
3. Wait for VSCode to load completely
4. `Cmd+Q` to quit vscode
5. **Quickly** back to terminal, up arrow, Enter to run the same command again

It only works when you're quick on step 4 and 5. Waiting after step 4 will not repro.

**Comment by agileVtalal** [comment]
Exiting vscode manually, `pkill code` and starting vscode again resolved the issue for me

**Comment by LucienMP** [comment]
> For me, what worked was to delete the folder `Service Worker` under `~/.config/Code` and re-opening my Visual Studio and it started working as expected. Hope it helps someone :)

This is still a problem with 1.64 just released, CentOS8/RHEL8, and moving from 1.61 to 1.64 but the ~/.config/Code folder is much older.

This solution worked for me

**Comment by RovoMe** [comment]
I just run into this issue while working on a new extension after updating vscode to the most recent version. So far I haven't experienced this issue and the code was more or less unmodified since yesterday where I extensively tested all changes.

![vscode-version](https://user-images.githubusercontent.com/1712331/152565898-fa2344c1-de0a-477b-87c5-b08c50d4e105.png)

I've updated vs-code today via the `.deb` package provided on the official site. Before I had it installed via snap but I found multiple comments that mentioned to use the official `.deb` package in favor of the snap one on Ubuntu systems.

What I figured out is that when I run the extension via the `Run Extension` debug option, every second restart of vscode (not the child environment started via the `Run Extension` command but the actual development editor!) all Webviews work as expected within the child editor, but every odd restart they do not. To me this sounds like something in regards to Webviews is cached here and not fully restorable on a reload which is then removed and hence works on every 2nd restart

**Comment by thelumiereguy** [comment]
> Exiting vscode manually, running `pkill code` and starting vscode again resolved the issue for me

Same here! Thanks!

**Comment by zeroxia** [comment]
I have never encountered this issue on macOS or Windows before.
But with Linux Ubuntu 18.04, this issue is not rare.
Just had this problem when requesting `Markdown: open preview to the side (Ctrl+K V)`,
then quit all VSCode instances and start again the preview is rendered.

VSCode version:
```
Version: 1.64.2
Commit: f80445acd5a3dadef24aa209168452a3d97cc326
Date: 2022-02-09T22:02:29.527Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.4.0-100-generic
```

I believe this should be the "standard version":
```shell
$ cat /etc/apt/sources.list.d/vscode.list 
### THIS FILE IS AUTOMATICALLY CONFIGURED ###
# You may comment out this entry, but any other modifications may be lost.
deb [arch=amd64,arm64,armhf] http://packages.microsoft.com/repos/code stable main
```

```shell
$ code --list-extensions
Conflict in settings file file:///home/zlxia/.config/Code/User/settings.json: Ignoring clang-format.executable.linux as clang-format.executable is ""
Conflict in settings file file:///home/zlxia/.config/Code/User/settings.json: Ignoring clang-format.executable.osx as clang-format.executable is ""
Conflict in settings file file:///home/zlxia/.config/Code/User/settings.json: Ignoring clang-format.executable.windows as clang-format.executable is ""
alefragnani.Bookmarks
azemoh.one-monokai
BazelBuild.vscode-bazel
cschlosser.doxdocgen
dbaeumer.vscode-eslint
eamodio.gitlens
GitHub.copilot
golang.go
jebbs.plantuml
jeff-hykin.better-cpp-syntax
llvm-vs-code-extensions.vscode-clangd
mark-wiemer.vscode-autohotkey-plus-plus
mjohns.clang-format
ms-python.python
ms-python.vscode-pylance
ms-toolsai.jupyter
ms-toolsai.jupyter-keymap
ms-vscode-remote.remote-containers
ms-vscode-remote.remote-ssh
ms-vscode-remote.remote-ssh-edit
ms-vscode-remote.remote-wsl
ms-vscode.cmake-tools
ms-vscode.cpptools
ms-vscode.cpptools-themes
ms-vscode.powershell
streetsidesoftware.code-spell-checker
twxs.cmake
vadimcn.vscode-lldb
vscodevim.vim
zero-plusplus.vscode-autohotkey-debug
zxh404.vscode-proto3
```

**Comment by viluon** [comment]
Just updated to 1.64.2 and was getting this issue with any webview (release notes, UI settings, extensions pages, ...). `pkill code` and starting again fixed this.

**Comment by V0XNIHILI** [comment]
Just had this on Ubuntu 20.04:

```Version: 1.65.0
Commit: b5205cc8eb4fbaa726835538cd82372cc0222d43
Date: 2022-03-02T11:12:36.248Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.13.0-30-generic snap```

For me (for now at least) restarting worked.

**Comment by jpaddeo** [comment]
https://github.com/microsoft/vscode/issues/125993#issuecomment-999715355

> For me, what worked was to delete the folder `Service Worker` under `~/.config/Code` and re-opening my Visual Studio and it started working as expected. Hope it helps someone :)

It works for me! I'm under Ubuntu 20.04.4 LTS ...thanks!

**Comment by fede-r1c0** [comment]
> Exiting vscode manually, running `pkill code` and starting vscode again resolved the issue for me
Same! thanks!

**Comment by fede-r1c0** [comment]
> Exiting vscode manually, running `pkill code` and starting vscode again resolved the issue for me

this has worked for me on manjaro linux

**Comment by elmarsto** [comment]
Worked for me as well. Ubuntu 20.04. Which, ironically, I am just now able to use again for work

**Comment by duaneking** [comment]
> > Exiting vscode manually, running `pkill code` and starting vscode again resolved the issue for me
> 
> this has worked for me on manjaro linux

Using just pure `pkill code` fixed this on ubuntu for me just now.  Clearly there is a defect here.

`~$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.4 LTS"
`

**Comment by xbbkok** [comment]
> For information: restarting the PC solved the issue for me ATM.
> 
> So, it may be related to some sort of cache that is cleared after reboot?
> 
> This issue appeared for me twice, every time after a vs code update. The first time updating from 1.55 to 1.56 and now updating from 1.56 to 1.57. I don't know if the fix on 1.56 happened after a reboot as well.

Thanks, it works

**Comment by liubomyr123** [comment]
I also have this issue on Linux. (Version VScode: 1.66.0)
Rebooting of computer fixed the issue, thanks!

**Comment by BreezeShane** [comment]
I meet this issue on Arco Linux by accident recently. It works well formerly.
The version of VScode is 1.66.0-1. And the image source is arcolinux_repo_xlarge.
```
arcolinux_repo_xlarge/visual-studio-code-bin 1.66.0-1 (94.7 MiB 281.1 MiB) (Installed)
    Visual Studio Code (vscode): Editor for building and debugging modern web and cloud applications (official binary version)
```

Anyway, I also found that VScode would have no response from time to time while I was pressing Backspace or Enter Key. I found it even more obvious when I tried to enter the path to my picture using `![xxx](...)`.

When I opened Preview Card, I found the log updated. Here is the new log:
```
[2022-04-10 16:34:33.495] [exthost] [error] [yzhang.markdown-all-in-one] provider FAILED
[2022-04-10 16:34:33.495] [exthost] [error] TypeError: Cannot read properties of undefined (reading 'document')
	at f (/home/user/.vscode/extensions/yzhang.markdown-all-in-one-3.4.0/dist/extension.js:1:296467)
	at p (/home/user/.vscode/extensions/yzhang.markdown-all-in-one-3.4.0/dist/extension.js:1:293567)
	at g (/home/user/.vscode/extensions/yzhang.markdown-all-in-one-3.4.0/dist/extension.js:1:295259)
	at v.provideCodeLenses (/home/user/.vscode/extensions/yzhang.markdown-all-in-one-3.4.0/dist/extension.js:1:298023)
	at o.provideCodeLenses (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:85:96644)
	at /opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:85:124517
	at Q._withAdapter (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:85:123100)
	at Q.$provideCodeLenses (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:85:124495)
	at i._doInvokeHandler (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:89:13826)
	at i._invokeHandler (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:89:13510)
	at i._receiveRequest (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:89:12119)
	at i._receiveOneMessage (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:89:10841)
	at /opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:89:8947
	at u.invoke (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:145)
	at v.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:1856)
	at d.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:19034)
	at /opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:104:34275
	at u.invoke (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:145)
	at v.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:1856)
	at d.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:19034)
	at o._receiveMessage (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:23615)
	at /opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:21149
	at u.invoke (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:145)
	at v.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:1856)
	at v.acceptChunk (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:15865)
	at /opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:14995
	at Socket.P (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:104:13797)
	at Socket.emit (node:events:390:28)
	at addChunk (node:internal/streams/readable:315:12)
	at readableAddChunk (node:internal/streams/readable:289:9)
	at Socket.Readable.push (node:internal/streams/readable:228:10)
	at Pipe.onStreamRead (node:internal/stream_base_commons:199:23)
```

Finally, my system info:
```
OS: ArcoLinux 
Kernel: 5.17.1-arch1-1 
Uptime: 28 mins 
Packages: 1580 (pacman) 
Shell: zsh 5.8.1 
Resolution: 1920x1080, 1920x1080, 1080x1920 
WM: awesome 
Theme: Arc-Dark [GTK2/3] 
Icons: Sardi-Arc [GTK2/3] 
Terminal: urxvt 
Terminal Font: MesloLGS NF-20 
CPU: Intel i5-10210U (8) @ 4.200GHz 
GPU: Intel CometLake-U GT2 [UHD Graphics] 
GPU: NVIDIA GeForce MX350 
Memory: 5637MiB / 15742MiB (35%) 
```

It would be kind of you to finish my issue, thank you very much!

**Comment by jafarijason** [comment]
If you close and re open vs code it will solve

**Comment by milnomada** [comment]
> Exiting vscode manually, running `pkill code` and starting vscode again resolved the issue for me

This the only one that fixed it for Osx 10.13.6

**Comment by nosratullah** [comment]
I'm using a Linux system and after upgrading to Vscode v 1.67 I faced the problem. Fortunately, rebooting solved my problem as mentioned in most of the comments!

**Comment by jramsay** [comment]
Hi @mjbvz : a few of us on the Live Share team have been reproing this quite a bit on Windows recently too.

Version: 1.67.2 (user setup)
Commit: c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5
Date: 2022-05-17T18:15:52.058Z
Electron: 17.4.1
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Windows_NT x64 10.0.18363

We use a webview for our session chat and started to see this intermittently when we attempt to open the Live Share Chat window. Verified that when this occurs it is for all webviews. Even if we disable Live Share they are broken. As others have mentioned we have to close all instances of VS Code to get out of this state. 

Is there any additional info we can collect for you the next time it occurs?

![image](https://user-images.githubusercontent.com/8161247/170349425-beef0ff6-d204-475d-ad1e-86b2176e3727.png)

**Comment by renato-bohler** [comment]
Just FYI, this happened to me on Linux (`pkill code` solved it for me), but I also managed to reproduce this on a bare VSCode for the web:


https://user-images.githubusercontent.com/25781956/173086228-babde1ab-a29c-4115-8cf6-0e4a68d3714d.mp4

**Comment by lovettchris** [comment]
I also get this on windows easily, using the [VSCode Lean4 extension](https://www.youtube.com/watch?v=zyXtbb_eYbY) which also has a webview - by simply running "code ." too quickly.  If I let the previous VS code fully shutdown then the next "code ." works fine.  But I have to give it 10 seconds or so.  It would be nice if vscode would fix this as it is very annoying.

**Comment by DuncanFaulkner** [comment]
this is still an issue in 1.70 on ubuntu 20.4

**Comment by c3-lukelewis** [comment]
Confirming that this is still an issue in version 1.71. Running MacOS Monterey 12.3.1

**Comment by Denovocto** [comment]
Still an Issue in 1.71 under Fedora 36
# Extensions:

> Graphviz Interactive Preview
> Version: 0.3.2
> VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=tintinweb.graphviz-interactive-preview

> Name: Graphviz (dot) language support for Visual Studio Code
> Version: 0.0.6
> VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=joaompinto.vscode-graphviz

Tested both extensions individually and getting same error message.
# vscode info

Version: 1.71.0
Commit: 784b0177c56c607789f9638da7b6bf3230d47a8c
Date: 2022-09-01T07:25:10.472Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Linux x64 5.19.4-200.fc36.x86_64

**Comment by shaperilio** [comment]
I get this in Windows when opening Jupyter notebooks. It's intermittent, and closing VSCode and reopening "fixes" it. Once you hit the error, though, no notebook will open until you restart.

Restarting is not really a fix since that can kill active kernels...

Version: 1.71.0 (user setup)
Commit: 784b0177c56c607789f9638da7b6bf3230d47a8c
Date: 2022-09-01T07:36:10.600Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19043
Sandboxed: No

**Comment by wrouesnel** [comment]
Also encountering this with the `git-graph` plugin on Ubuntu 22.04:

Version: 1.71.2
Commit: 74b1f979648cc44d385a2286793c226e611f59e7
Date: 2022-09-14T21:12:14.256Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Linux x64 5.15.0-27-generic
Sandboxed: No

**Comment by andresmitre** [comment]
Restaring worked for me, I also re-installed jupyter notebook extension!

**Comment by Productivix** [comment]
same problem : no display of jpg images on Linux
$ code --version
1.73.1
6261075646f055b99068d3688932416f2346dd3b
x64
$ cat /etc/*release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"

message : 
Error loading webview: Error: Could not register service workers: InvalidStateError: 
Failed to register a ServiceWorker: The document is in an invalid state..

**Comment by apupier** [comment]
Since yesterday, I encountered the same issue for all webviews, tried with: the one in Walkthrough, Quarkus welcome page, Kaoto UI.

`Error loading webview: Error: Could not register service workers: InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state..`

I am on Fedora 36
VS Code 1.73.1

I tried restating VS Code, restarting OS. Still same behavior.
I tried with another fresh instance of VS Code.

**Comment by beyond9thousand** [comment]
bruh, is no one ever going to fix this?

**Comment by tonescones** [comment]
Please fix this <3

**Comment by drexler** [comment]
I had the same issue with Mac OS 13.0.1 running   VS Code `1.73.1` .   Putting my resolution out here for anyone how needs it.
`sudo pkill code` and then force close any VS Code instances.  Thanks @milnomada

**Comment by binarycache** [comment]
If anyone is using Brave, please turn off the brave shields as well.

**Comment by rutvik-simformsolutions** [comment]
i am getting the same issue for [Laravel Artisan Extension](https://marketplace.visualstudio.com/items?itemName=ryannaddy.laravel-artisan)
![image](https://user-images.githubusercontent.com/97081188/207653716-8105e295-e07e-4816-9bfc-fe47ec3a7e81.png)

**Comment by AlexZeitler** [comment]
I'm facing the same error for the internal Markdown preview as well as the Markdown Enhanced extension on Windows.

**Comment by hummerdi** [comment]
It is still here. Sad...

```Version: 1.74.2
Commit: e8a3071ea4344d9d48ef8a4df2c097372b0c5161
Date: 2022-12-20T10:27:19.253Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Linux x64 5.15.0-56-generic
Sandboxed: No
```

**Comment by tkossak** [comment]
Same here on Manjaro Linux and vscode 1.74.2-1 installed from AUR.
Killing all `code` processes is a workaround meantime.

**Comment by adizhol** [comment]
This happens after every update from Snap on Ubuntu 20.04 
please fix this...

**Comment by A404coder** [comment]
It really bothers me, so many features rely on this.
I even can't see release Notes locally.🤯
MacOS 12.5.1
VS code 1.74.2

**Comment by hgehlhausen** [comment]
I have a known solution on Ubuntu 22.04
- From analysis, this issue is definitely caused by the ~/.config/Code folder contents.
I deleted all of my extensions, and still experienced the issue with **Core Markdown Preview Functionality**

I deleted my "~/.config/Code" folder and was able to restore functionality. Its going to be a pain installing all my extensions again, but I've narrowed down the error ( at least in my system )

You can *probably* fix the issue by clearing the Code folder, or by clearing your "settings". What I suspect is that VS code has some older config-settings which break when updating the version, or become corrupted by some other process.

**Comment by Leigh-M** [comment]
I am seeing this when trying to view the latest Release Notes:

Error loading webview: Error: Could not register service workers: InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state..

**Comment by mrliptontea** [comment]
I just hit this issue with the latest release trying to display the Release Notes.
I deleted `~/.config/Code/Service Worker` (+ all directories with `Cache` in the name) and now it's working again.

**Comment by Leigh-M** [comment]
Restarting my machine worked as well..  the old faithful turn it off an on again..! ;-)

**Comment by UlisseMini** [comment]
`rm -rf .config/Code/Service\ Worker/{CacheStorage,ScriptCache}` followed by `pkill code` worked for me without restarting my whole machine.

**Comment by fbarikzehi** [comment]
> I found that the problem ca be fixed without rebooting by removing all the cache folders used by vscode:
> 
>     * close vscode (make sure no process is running)
> 
>     * `rm -rf ~/.config/Code/*Cache*`
> 
>     * open vs code

It works for me.

**Comment by chris-aeviator** [comment]
I'm affected by this on linux, not even the markdown preview works. I have a lot of `[code] <defunct>` entries in my `ps -ef` list. `killall code` won't remove them (sam with `pkill`). Restarting your computer to fix your code editor is such a microsofty thing and should never be asked from linux users :smiley:

**Comment by qadram** [comment]
I'm not sure why #178834 was closed, because this issue is happening when updating to 1.77, and 1.76.2 worked perfectly. I have been checking and this change should have caused the issue:

https://github.com/microsoft/vscode/pull/177716/commits/f71995e0b16ec2429cbbd221714984197ff5259d#diff-2f29d5590142a69ba7b8500a64b19f6bffb00b3b6bace7329976f6e5fea88649

My extension is totally broken because the webview cannot show my web app and it's kind of difficult to send you a reproducible test case.

**Comment by starball5** [comment]
Possibly related on Stack Overflow: [Error loading webview: Error: Could not register service workers: TypeError: Failed to register a ServiceWorker for scope](https://stackoverflow.com/q/67698176/11107541)

**Comment by Sberm** [comment]
On windows that's %AppData%\Code\Cache

**Comment by rdking** [comment]
Also happens in 1.80.0 and 1.80.1 with "Hex Editor" installed and trying to open a binary file.

**Comment by DiguyaDeveloper** [comment]
> rm -rf .config/Code/Service\ Worker/{CacheStorage,ScriptCache}

`pkill code
rm -rf .config/Code/Service\ Worker/{CacheStorage,ScriptCache}`

Obrigado, essa solução funcionou para mim, estou usando a versão 23.x do ubuntu

**Comment by bryanmramsamy** [comment]
I still have the issue and rebooting or restarting VS Code over and over again works, but is extremely time-consuming and therefore not a viable option.

**Comment by BadLice** [comment]
I'm still having this issue running GitLens commands. I'm on Ubuntu

**Comment by ssibrahimpur** [comment]
Issue still persist on latest version

![image](https://github.com/microsoft/vscode/assets/47515317/370939d0-a8b5-4a4a-9475-b0132bdcbc42)

**Comment by jkoubs** [comment]
The issue appeared to me when trying updating a new vscode version.
Solution: Reboot pc, thanks @tritemio

**Comment by enkeyz** [comment]
For me it occurs anytime I install an extension, which opens up their setting page right after install.

**Comment by tp1050** [comment]
> `rm -rf .config/Code/Service\ Worker/{CacheStorage,ScriptCache}` followed by `pkill code` worked for me without restarting my whole machine.

worked like a charm.

**Comment by wafflemakr** [comment]
> I just hit this issue with the latest release trying to display the Release Notes. I deleted `~/.config/Code/Service Worker` (+ all directories with `Cache` in the name) and now it's working again.

This worked like a charm for me in Ubuntu, remember to refresh VSCode! thanks a lot!

**Comment by Tyriar** [comment]
This isn't a linux/snap issue, I've seen it plenty of times on Windows.

**Comment by QuentinSc** [comment]
Get it on windows, on a fresh install

**Comment by saadazghour** [comment]
If you are using Ubuntu, there is probably another (maybe hidden) vscode process, which is causing the problem.

Close the vs code first and in terminal try:   `killall code`

**Comment by mminot-yseop** [comment]
Got that all of a sudden on 1.87.0. OK after reboot but I hope it won’t come back too often. 😆

**Comment by fabriceleal** [comment]
This could be a blink bug, maybe resulting from some sort of race condition ... the error is triggered here https://github.com/chromium/chromium/blob/main/third_party/blink/renderer/modules/service_worker/service_worker_container.cc#L427-L430 

The `provider_` might not be set at creation 
https://github.com/chromium/chromium/blob/main/third_party/blink/renderer/modules/service_worker/service_worker_container.cc#L176-L181

**Comment by juanma-avio** [comment]
**Easy workaround on Linux:**
1. exit vs code
2. in the console write: "killall code"
3. start vs code again

**Comment by caiqichang** [comment]
i got this problem when i open second  vscode windows.
only first vscode window can work well, webview in the other windows broken down.

vscode version: 1.89.0
pc: windows 11

**Comment by yan-pi** [comment]
Same issue Here!!

Does this issue occur when all extensions are disabled?: No, sice i need to install the markdown extension to occur

VS Code Version: 1.89.1 https://github.com/microsoft/vscode/commit/dc96b837cf6bb4af9cd736aa3af08cf8279f7685 x64
OS Version: linux fedora 40 linux kernel 6.8.9-300.fc40.x86_64
Steps to Reproduce:

Install markdown preview or other Rxtension that uses webview
Run the command to show webview

**Comment by BMourguesFieldbox** [comment]
> > `rm -rf .config/Code/Service\ Worker/{CacheStorage,ScriptCache}` followed by `pkill code` worked for me without restarting my whole machine.
> 
> worked like a charm.

Same, worked perfectly,  quote replying for future readers

**Comment by Doc-Tiebeau** [comment]
> > `rm -rf .config/Code/Service\ Worker/{CacheStorage,ScriptCache}` followed by `pkill code` worked for me without restarting my whole machine.
> 
> worked like a charm.

Works for same issue when displaying markdown and png files.

OS: Debian Bookworm 12.6
VScodium : 1.91.1
Release : 24193
Context : ~/.config/VSCodium synchronized using rsync from previous machine (Ubuntu + Snap)

**Comment by CHN-STUDENT** [comment]
Today i try to preview markdown file by using VSCode on my windows computer, i get this program. I use `taskkill /F /IM "Code.exe" /T` to kill all vscode process, now seem is work normally.

**Comment by sammcj** [comment]
Had this today after upgrading VSCode, the fix was to delete `~/Library/Application\ Support/Code/Service\ Worker`

**Comment by jaens** [comment]
Had the same issue today with the latest VSCode version.

Note that you really need to kill **all** VSCode processes as a workaround. 

Even though I am on Linux and run VSCode in a systemd cgroup (service), killing that was not enough, since apparently there's some bug now in using `code -w <file>`, and a zombie `code` process outside the cgroup had hung (requiring SIGKILL).

(process tree:)
```
\_ /usr/share/code/code -w --waitMarkerFilePath /tmp/U6r5u4r5
|   \_ [code] <defunct>
|   \_ [code] <defunct>
|   \_ [code] <defunct>
|   \_ [code] <defunct>
|   \_ [code] <defunct>
```

**Comment by AlexanderZagaynov** [comment]
### Additional Linux case: VS Code 1.128.0, webview recovers only after resetting Service Worker state

I am seeing a recurring blank Codex webview on Linux. The activity-bar icon blinks when selected, but the panel never renders.

#### Environment

- Fedora Silverblue 44
- Kernel 7.1.3-200.fc44.x86_64
- VS Code 1.128.0
- Commit `fc3def6774c76082adf699d366f31a557ce5573f`
- Architecture: x64
- Webview provider: `openai.chatgpt` 26.707.31428

#### Broken startup

The extension host and its backend initialized normally:

```text
Activating Codex extension
Spawning codex app-server
Initialize received id=1
```

However, the webview never logged either of its normal renderer-startup markers:

```text
React root render requested
app routes mounted
```

There was no extension reinstall, authentication change, configuration change, or modification to `~/.codex`.

#### Controlled recovery

I fully stopped all VS Code processes and moved only:

```text
~/.config/Code/Service Worker
```

to a backup. I then restarted the same VS Code build, extension build, profile, and workspace.

The webview immediately initialized successfully:

```text
Activating Codex extension
Initialize received id=1
React root render requested windowType=extension
app routes mounted
```

This provides a controlled before/after comparison: resetting VS Code's shared Service Worker state was the only relevant change.

#### Recurrence and retained artifacts

This has happened at least twice. I retained two Service Worker directories from separate recoveries, sized 84 KB and 60 KB.

The small sizes suggest this is not simply caused by unbounded cache growth. The latest saved database contains registrations for VS Code webview origins, but its LevelDB logs contain no explicit corruption message.

I did not capture the webview Developer Tools console before recovery, so I cannot confirm the exact `InvalidStateError` string.

I retained the failing Service Worker directory locally and can extract additional non-sensitive metadata if useful.

**Comment by szarnekow** [comment]
Maybe this helps to narrow it down: the stuck state seems to be scoped to individual webview origins, not something that clears globally when VS Code updates.

Each webview's origin is derived from `base32(sha256({parentOrigin, salt}))`, where salt is a UUID minted once per (extensionId, viewType) and stored in memento/mainThreadWebviewPanel.origins. I decoded that against the local Service Worker/Database/*.log on a real machine (1.135.0, so the current service-worker.js is VERSION = 6), and some origins were cleanly on v=6, while others from the same extension were still on v=4 or v=5 and just never moved forward. That lines up with what people keep describing here: some extensions/views work fine while others don't, or something that worked breaks again after a later update.

I also tried to see if a webview could just unregister and re-register itself, but getRegistration()/getRegistrations() hit the same guard that register() does, so a document sitting on a wedged origin cannot list its own registrations. The only thing that actually works is fully quitting VS Code and wiping the whole Service Worker folder, which is a pretty blunt instrument since it throws away every origin's state, not just the broken one.

Given the problem looks per-origin, and session.clearStorageData({storages: ['serviceworkers','cachestorage']}) already exists, would something like a `Developer: Reset Webview Service Worker command` be doable? Clear just the current webview's origin and reload it. That'd give people an actual recovery path instead of "delete a folder and hope."

**Comment by dnch13** [comment]
Have the same problem - i bet would be reasonable to create at-least some retry mechanism with cache cleanup until the normal solution is found, beacause it eats up TOO MUCH time from all people around the globe using VSCode, so something should be done immediately.

Here's what i have: "Error loading webview: Error: Could not register service worker: InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state.."
